content-src/reducers/reducers.js

```diff
@@ -21,8 +21,34 @@ function setRowsOrError(type) {
   };
 }
 
+function setSearchState(type) {
+  return (prevState = {currentEngine: {}, error: false}, action) => {
+    const state = {};
+    switch (action.type) {
+      case am.type(type):
+        if (action.error) {
+          state.currentEngine = {};
+          state.error = action.data;
+        } else {
+          state.error = false;
+          state.currentEngine = {
```

Comment:
I think it would be a good idea to have a check here for `action.data` and `action.data.currentEngine` or a try/catch in case something goes unexpectedly wrong.

Also, you're probably safe to pass `action.data.currentEngine`, since actions are plain objects